### PR TITLE
fix(Row): checkbox onChange always returning true

### PR DESCRIPTION
**Pull request by RubenMorPov**
[WEB-460](https://jira.tid.es/browse/WEB-460) Fixed onChange parameter, now passes "isChecked".

### CODE REVIEW

File: src/list.tsx [hunk 1]
```diff
@@ -374,7 +374,7 @@ const useControlState = ({
             setIsChecked(!isChecked);
         }
         if (onChange) {
-            onChange(!value);
+            onChange(!isChecked);
```
Comment (atabel) [inline src/list.tsx:377]: I think this should be:
```ts
onChange(isControlledByParent ? !value : !isChecked)
```
Please, add some tests to check both cases (controlled by parent and uncontrolled)

**Review by atabel — CHANGES_REQUESTED** [review]

File: src/list.tsx (hunk 1 above)
Comment (pladaria) [reply src/list.tsx:377]: in this case some simple unit tests should be enough

File: src/__tests__/list-test.tsx [hunk 2]
```diff
@@ -37,10 +37,16 @@ test('Row as a button', () => {
 });
 
 test('Row with switch', () => {
+    const onChangeFn = (isChecked?: boolean) => {
+        return isChecked;
+    };
+
+    const spyOnChange = jest.fn(onChangeFn);
```
Comment (pladaria) [inline src/__tests__/list-test.tsx:44]: you can write this as:
```js
const onChangeSpy = jest.fn((isChecked?: boolean) => {}); // The callback doesn't need to return anything
```

File: src/__tests__/list-test.tsx [hunk 3]
```diff
@@ -37,10 +37,16 @@ test('Row as a button', () => {
 });
 
 test('Row with switch', () => {
+    const onChangeFn = (isChecked?: boolean) => {
+        return isChecked;
```
Comment (pladaria) [inline src/__tests__/list-test.tsx:41]: no need to return anything

**Review by pladaria — APPROVED** [review]

File: src/__tests__/list-test.tsx [hunk 4]
```diff
@@ -37,10 +37,16 @@ test('Row as a button', () => {
 });
 
 test('Row with switch', () => {
```
Comment (pladaria) [inline src/__tests__/list-test.tsx:39]: Looks like the test is not passing
image

**Review by atabel — APPROVED** [review]

**Pull request merged**